sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -321,8 +338,30 @@ static void _transmit_pkt(gnrc_lorawan_t *mac)
         last_snip = last_snip->iol_next;
     }
 
+    uint32_t channel = gnrc_lorawan_pick_channel(mac);
+    (void)channel;
+
+    uint16_t conf_fcnt = 0;
+    if (IS_ACTIVE(CONFIG_GNRC_LORAWAN_1_1)) {
+        for (unsigned i = 0; i < GNRC_LORAWAN_MAX_CHANNELS; i++) {
+            if (mac->channel[i] == channel) {
+            mac->last_chan_idx = i;
+            }
```

Comment:
since you are already there: maybe at this point it makes sense to simply change the `gnrc_lorawan_pick_channel` function to return the channel index. You can e.g add another helper function to get the channel frequency from the channel index. That way you don't need to run this loop again to find the channel index.